smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/ServiceConfig.kt

```diff
@@ -8,25 +8,27 @@ package software.amazon.smithy.swift.codegen.integration
 import software.amazon.smithy.codegen.core.Symbol
 import software.amazon.smithy.swift.codegen.ClientRuntimeTypes
 import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.utils.toPascalCase
 
 /**
  * Represents a config field on a client config struct.
  */
-data class ConfigField(val memberName: String?, val type: Symbol, val formatter: String = "\$N", private val documentation: String? = null)
+data class ConfigField(val memberName: String?, val type: Symbol, val propFormatter: String = "\$N", private val documentation: String? = null, val paramFormatter: String = "\$D")
 
 /**
  * ServiceConfig abstract class that allows configuration customizations to be configured for the protocol client generator
  */
 abstract class ServiceConfig(val writer: SwiftWriter, val serviceName: String) {
```

Comment:
**Question**: Why is this `abstract`? The only implementation I can see in this PR is `DefaultServiceConfig`. Are there (or will there be) different implementations?

Reply:
Yes, there are two implementations of it. `DefaultServiceConfig` for generic codegen and `AWSServiceConfig` for AWS codegen.

https://github.com/awslabs/aws-sdk-swift/blob/5496c694c80fcfd0b3d825af632a55a677251f8a/codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSServiceConfig.kt#L40